connect/runtime/src/main/java/org/apache/kafka/connect/runtime/SinkConnectorConfig.java

```diff
@@ -42,9 +43,19 @@
     public static final String TOPICS_REGEX_DEFAULT = "";
     private static final String TOPICS_REGEX_DISPLAY = "Topics regex";
 
+    public static final String DLQ_PREFIX = "errors.deadletterqueue.";
+
+    public static final String DLQ_TOPIC_NAME_CONFIG = DLQ_PREFIX + "topic.name";
+    public static final String DLQ_TOPIC_NAME_DOC = "The name of the topic to be used as the dead letter queue (DLQ) for messages that " +
+        "result in an error when processed by this sink connector, transformations, or converter. The topic name is blank by default, " +
+        "which means that no messages are to be recorded in the DLQ.";
+    public static final String DLQ_TOPIC_DEFAULT = "";
+    private static final String DLQ_TOPIC_DISPLAY = "DLQ Topic Name";
+
     static ConfigDef config = ConnectorConfig.configDef()
         .define(TOPICS_CONFIG, ConfigDef.Type.LIST, TOPICS_DEFAULT, ConfigDef.Importance.HIGH, TOPICS_DOC, COMMON_GROUP, 4, ConfigDef.Width.LONG, TOPICS_DISPLAY)
-        .define(TOPICS_REGEX_CONFIG, ConfigDef.Type.STRING, TOPICS_REGEX_DEFAULT, new RegexValidator(), ConfigDef.Importance.HIGH, TOPICS_REGEX_DOC, COMMON_GROUP, 4, ConfigDef.Width.LONG, TOPICS_REGEX_DISPLAY);
+        .define(TOPICS_REGEX_CONFIG, ConfigDef.Type.STRING, TOPICS_REGEX_DEFAULT, new RegexValidator(), ConfigDef.Importance.HIGH, TOPICS_REGEX_DOC, COMMON_GROUP, 4, ConfigDef.Width.LONG, TOPICS_REGEX_DISPLAY)
+        .define(DLQ_TOPIC_NAME_CONFIG, ConfigDef.Type.STRING, DLQ_TOPIC_DEFAULT, Importance.MEDIUM, DLQ_TOPIC_NAME_DOC, ERROR_GROUP, 6, ConfigDef.Width.MEDIUM, DLQ_TOPIC_DISPLAY);
```

Comment:
Should this be added to the Errors group?

Reply:
I am ok to put the property in this group. The only worrisome point is the hardcoded `6` here. Any way to get around this?

Reply:
I don’t know of a way to do that.

Reply:
Well, one way is to iterate through the ConfigKeys and count the number of existing keys in the group. But there's no existing utility or method to do this.

Reply:
Can we make orderInErrorGroup a protected static field in `ConnectorConfig` and use `++orderInErrorGroup` instead of `6`? Same for `orderInGroup`?

Reply:
That seems even more brittle, doesn't it. :-)

Reply:
Ok. I'll leave it as it is then. Let me know if you have any further comments.

Reply:
Yeah, ConfigDef doesn't break if there are duplicate `orderInGroup` values, and it tries to [order them in a somewhat orderly fashion](https://github.com/apache/kafka/blob/38e9958d6e5d2224fbb9a54d04b1505ad561e8a3/clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java#L1246-L1270). This is why I think this is less brittle than trying to manage those orders across multiple Configuration classes.

Reply:
Yeah, makes sense. That's how the `TOPICS_CONFIG` and `TOPICS_REGEX_CONFIG` work without any issues. Thanks, Randall.